nova_vm/src/ecmascript/types/spec/data_block.rs

```diff
@@ -61,18 +67,248 @@ mod private {
     impl Sealed for f64 {}
 }
 
-pub trait Viewable: private::Sealed {}
+pub trait Viewable: private::Sealed + Copy {
+    fn is_bigint_type() -> bool;
+    fn into_be_value(self, agent: &mut Agent) -> Value;
+    fn into_le_value(self, agent: &mut Agent) -> Value;
+    fn from_le_value(agent: &mut Agent, gc: GcScope<'_, '_>, value: Value) -> Self;
+    fn from_be_value(agent: &mut Agent, gc: GcScope<'_, '_>, value: Value) -> Self;
+}
```

Comment:
**note:** to and from ne (natural alignment of the machine) might be of use as well.